libgit/repo.go

```diff
@@ -846,17 +875,37 @@ func GCRepo(
 
 	// Check the GC thresholds again since they might have changed
 	// while getting the lock.
-	doPackRefs, numLooseRefs, err := needsGC(storer, options)
+	doPackRefs, numLooseRefs, doPruneLoose, err := needsGC(storage, options)
 	if err != nil {
 		return err
 	}
-	if !doPackRefs {
+	if !doPackRefs && !doPruneLoose {
 		log.CDebugf(ctx, "GC no longer needed")
 		return nil
 	}
 
-	// For now just pack refs.  TODO: add object pruning and object
-	// re-packing.
-	log.CDebugf(ctx, "Packing %s loose refs", numLooseRefs)
-	return storer.PackRefs()
+	if doPackRefs {
+		log.CDebugf(ctx, "Packing %s loose refs", numLooseRefs)
+		err = storage.PackRefs()
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
Wouldn't it be better to first prune and then pack. And not pack and then prune. (if prune deletes some objects then adding them to a pack is not good). I.e. just change the order of pruning and packing here?

Reply:
Above is just packing refs, not objects.  It won't ever delete or unreference any objects.  I don't think the order matters, but please correct me if I'm wrong.

Reply:
Sorry for the noice, you are right.

Reply:
Sorry for the noise, you are right.